app/helpers/patient_query_helper.rb

```diff
@@ -345,17 +347,39 @@ def advanced_filter(patients, filters, tz_offset)
       when 'manual-contact-attempts'
         # less/greater-than operators are flipped for where_assoc_count
         operator = :==
-        operator = :> if filter[:value][:operator] == 'less-than'
-        operator = :>= if filter[:value][:operator] == 'less-than-equal'
-        operator = :<= if filter[:value][:operator] == 'greater-than-equal'
-        operator = :< if filter[:value][:operator] == 'greater-than'
-        case filter[:value][:option]
+        operator = :> if filter[:numberOption] == 'less-than'
+        operator = :>= if filter[:numberOption] == 'less-than-equal'
+        operator = :<= if filter[:numberOption] == 'greater-than-equal'
+        operator = :< if filter[:numberOption] == 'greater-than'
+        case filter[:optionalOption]
         when 'Successful'
-          patients = patients.where_assoc_count(filter[:value][:number], operator, :contact_attempts, successful: true)
+          patients = patients.where_assoc_count(filter[:value], operator, :contact_attempts, successful: true)
         when 'Unsuccessful'
-          patients = patients.where_assoc_count(filter[:value][:number], operator, :contact_attempts, successful: false)
+          patients = patients.where_assoc_count(filter[:value], operator, :contact_attempts, successful: false)
         when 'All'
-          patients = patients.where_assoc_count(filter[:value][:number], operator, :contact_attempts)
+          patients = patients.where_assoc_count(filter[:value], operator, :contact_attempts)
+        end
+      when 'age'
+        # specific case where value is a range not a single value
+        if filter[:numberOption] == 'between'
+          low_age = filter[:value][:low].to_i
+          high_age = filter[:value][:high].to_i
+          patients = patients.where('date_of_birth > ?', DateTime.now - high_age.year).where('date_of_birth <= ?', DateTime.now - low_age.year)
```

Comment:
No it shouldn't... let me explain with an example because it is easier.  Say you filter age between 20 and 30 today 1/19/2021.  The youngest possible 20 year old turns 20 today, so that birthday is 1/19/2001. The oldest possible 30 year old will be turning 31 tomorrow, so their birthday is 1/20/1990. With this between filter you would want all monitorees with birthdays between 1/20/1990-1/19/2001.

`.where('date_of_birth > ?', DateTime.now - high_age.year)` does the high end of the filter (i.e. 30).  What it does is takes today's date 1/19/2021 and subtracts the high bound (30) plus one (so 31) in years from that date, which would be 1/19/1990.  We don't want to include 1/19/1990 because that person is 31, hence why this where statement does not have an equal value.

`.where('date_of_birth <= ?', DateTime.now - low_age.year)` does the high end of the filter (i.e. 20).  What this does is takes today's date 1/19/2021 and subtracts the lower bound (20) in years from that date, which would be 1/19/2001. This where statement will include every monitoree who has a birthday before 1/19/2001 INCLUDING this date.

When the two where statements are put together, it filters monitorees who have birthdays after 1/19/1990 (NOT including that day), but before 1/19/2001 (including this day).  Does that make sense why one side has an equality and the other doesn't?

Reply:
@hackrm I noticed I was off by one year on the high bound so that should be fixed now

Reply:
This is useful because I had the same gut reaction. Can you add a comment above these lines @sgober with a concise explanation of this?

Reply:
done!